libpod/container_internal.go

```diff
@@ -1009,7 +1081,12 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 		}
 	}
 
-	g.SetRootPath(c.state.Mountpoint)
+	mountpoint := c.getUserNSMountpoint()
```

Comment:
If we're using a separate mountpoint from the c/storage mountpoint, we should just set c.state.Mountpoint to the real one.

Reply:
done